app/controllers/subject_plan_subjects_controller.rb

```diff
@@ -0,0 +1,36 @@
+class SubjectPlanSubjectsController < ApplicationController
+  include SubjectsHelper
+
+  before_action :authenticate_user!
+
+  def index
+    json = not_planned_subjects.group_by(&:category).map do |category, subjects|
+      {
+        label: formatted_category(category),
+        id: category,
+        choices: subjects.map do |subject|
+          {
+            label: display_planned_subject_name_and_code(subject),
+            value: subject.id,
+          }
+        end
+      }
+    end
+
+    render json:
+  end
+
+  private
+
+  def not_planned_subjects
```

Comment:
En realidad estas serían materias no planeadas ni aprobadas, no?

```suggestion
  def not_planned_and_not_approved_subjects
```

Reply:
Aunque me cuestiono un poco si quizás podríamos simplemente mostrar las materias aprobadas en el select – la verdad no me parece para nada mal. Vos que opinás?

Reply:
estoy de acuerdo con el cambio a `not_planned_and_not_approved_subjects ` lo otro capaz lo podemos discutir en otro PR.